### PR TITLE
Array support

### PR DESCRIPTION
**Pull request by awegrzyn**
- [x] Move JSON file reading out of constructor
- [ ] Add] JSON <-> Consul unit test

### CODE REVIEW

**Comment by awegrzyn** [comment]
@Barthelemy @teo Can you take a look at the `configuration-convert` exec which I added in this PR, it allows converting JSON <-> Consul

File: src/Backends/Consul/ConsulBackend.cxx [hunk 1]
```diff
@@ -58,12 +88,41 @@ boost::optional<std::string> ConsulBackend::getString(const std::string& path)
 
 boost::property_tree::ptree ConsulBackend::getRecursive(const std::string& path)
 {
-  auto requestKey = replaceDefaultWithSlash(addPrefix(path));
+  auto requestKey = replaceDefaultWithSlash(addPrefix(path)) + "[]";
```
Comment (teo) [inline src/Backends/Consul/ConsulBackend.cxx:91]: Is `[]` appended to every request?

File: src/CommandLineUtilities/Convert.cxx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+#include "Configuration/ConfigurationFactory.h"
+#include <fstream>
+#include <boost/program_options.hpp>
+
+using namespace std::literals::string_literals;
+
+int main(int argc, char *argv[]) {
+  std::string sourceUri, destinationUri;
+  const std::string TEMP_FILE = "/tmp/alice_o2_configuration_convert_file.json";
```
Comment (teo) [inline src/CommandLineUtilities/Convert.cxx]: Why is a temporary file made necessary?

Comment (awegrzyn) [reply src/CommandLineUtilities/Convert.cxx]: Ah true, it's a work around of file reading in constructor, I'll get rid of it

File: src/Backends/Consul/ConsulBackend.cxx (hunk 1 above)
Comment (awegrzyn) [reply src/Backends/Consul/ConsulBackend.cxx:91]: This is the only way to have one method (`getRecursive`) providing trees and arrays. To get rid of it I could append interface with `getArray`.

**Pull request merged**